src/main/daml/Daml/Finance/Settlement/Instruction.daml

```diff
@@ -69,80 +67,112 @@ template Instruction
         assertMsg "actors must intersect with settlers" $ not $ S.null $ S.intersection actors settlers
+        let void allocation approval = abort $ "unsupported combination " <> show (allocation, approval) <> messageSuffix
         -- execute instruction
         case (allocation, approval) of
-          (Unallocated, _) -> abort $ "instruction must be allocated" <> messageSuffix
-          (_, Unapproved) -> abort $ "instruction must be approved" <> messageSuffix
-          (Pledge transferableCid, TakeDelivery receiverAccount) -> do
-            transferable <- fetch transferableCid
-            let senderAccount = getAccount transferable
-            transferableCid <- exercise transferableCid Transferable.Transfer with actors = signed; newOwnerAccount = receiverAccount
-            when undiscloseSenderAccount $ undiscloseA senderAccount $> ()
-            when undiscloseReceiverAccount $ undiscloseA receiverAccount $> ()
-            -- disclose to actors (such that they can get the TemplateTypeRep in the Batch)
-            Some <$> discloseT step.receiver actors transferableCid
-          (Pledge transferableCid, DebitSender) -> do
-            transferable <- fetch transferableCid
-            let senderAccount = getAccount transferable
-            assertMsg ("the custodian of the transferable must be the receiver" <> messageSuffix) $ senderAccount.custodian == step.receiver
-            accountCid <- Account.exerciseInterfaceByKey @Account.I senderAccount step.sender Account.Debit with holdingCid = coerceContractId transferableCid
-            when undiscloseSenderAccount $ undiscloseA senderAccount $> ()
-            pure None
-          (CreditReceiver, TakeDelivery receiverAccount) -> do
-            assertMsg ("the custodian of the receiver account must be the sender" <> messageSuffix) $ receiverAccount.custodian == step.sender
-            transferableCid : ContractId Transferable.I <- coerceContractId <$> Account.exerciseInterfaceByKey @Account.I receiverAccount step.receiver Account.Credit with quantity = step.quantity
-            when undiscloseReceiverAccount $ undiscloseA receiverAccount $> ()
-            -- disclose to actors (such that they can get the TemplateTypeRep in the Batch)
-            Some <$> discloseT step.receiver actors transferableCid
-          (CreditReceiver, DebitSender) -> do
-            assertMsg ("sender must be the same party as receiver" <> messageSuffix) $ step.sender == step.receiver
-            pure None
-          (SettleOffLedger, SettleOffLedgerAcknowledge) -> pure None
-          (SettleOffLedger, _) -> abort $ "offledger settlement must be acknowledged" <> messageSuffix
-          (_, SettleOffLedgerAcknowledge) -> abort $ "offledger settlement must be initiated" <> messageSuffix
+          (Unallocated, _) -> void allocation approval
+          (_, Unapproved) -> void allocation approval
+          (PassthroughFrom _, _) -> void allocation approval
```

Comment:
Why is this not a valid case? Is it because we have already "trimmed" the graph and killed all the `PassThroughFrom`?

Reply:
yes, exactly. the ordering of the instructions are such that the `PassthroughFrom ..`-instructions are being allocated with a `Pledge passthroughCid` before it is being executed itself.